frontend/checks-runner.ts

```diff
@@ -0,0 +1,160 @@
+
+// Keep track of current child process
+let currentChildProcess: ReturnType<typeof spawn> | null = null;
+
+// Handle parent process termination
+process.on('SIGINT', () => {
+  console.log(chalk.yellow('\nGracefully shutting down...'));
+  cleanup();
+  process.exit(0);
+});
+
+process.on('SIGTERM', () => {
+  console.log(chalk.yellow('\nGracefully shutting down...'));
+  cleanup();
+  process.exit(0);
+});
+
+// Cleanup function to kill child process
+function cleanup() {
+  if (currentChildProcess) {
+    console.log(chalk.yellow('Terminating running scripts...'));
+    currentChildProcess.kill('SIGTERM');
+    currentChildProcess = null;
+  }
+}
+
+// Resolve the path to package.json
+const packagePath = path.resolve(process.cwd(), 'package.json');
+
+// Check if package.json exists
+if (!fs.existsSync(packagePath)) {
+  console.error(chalk.red('Error: package.json not found in the current directory.'));
+  process.exit(1);
+}
+
+// Read and parse package.json
+const packageJson: PackageJson = JSON.parse(fs.readFileSync(packagePath, 'utf8'));
+const scripts = Object.keys(packageJson.scripts ?? {});
+
+// Check for missing allowed scripts in package.json
+const missingAllowedScript = allowedScripts.filter(({ value: name }) => !scripts.includes(name));
+
+if (missingAllowedScript.length > 0) {
+  console.error(
+    chalk.red('Error: allowed script(s) ["%s"] not found in package.json.'),
+    missingAllowedScript.map(({ value: name }) => name).join('", "'),
+  );
+  process.exit(1);
+}
+
+// Filter the allowed scripts that are present in package.json
+const scriptList = allowedScripts.filter(({ value: name }) => scripts.includes(name));
+
+// Function to run selected scripts
+async function runScripts() {
+  if (scriptList.length === 0) {
+    console.log(chalk.yellow('No scripts found in package.json.'));
+    return;
+  }
+
+  try {
+    // Prompt user to select scripts to run
+    const answers = await checkbox({
+      message: 'Select the scripts to run (in order):',
+      choices: scriptList.flatMap(({ value, checked, name, description, seperator }) => [
+        ...(seperator ? [new Separator(chalk.gray(`  --- ${seperator} ---`))] : []),
+        { value, name, description, checked },
+      ]),
+      loop: false,
+      required: true,
+      pageSize: 10,
+    });
+
+    // Execute each selected script
+    for (const answer of answers) {
+      const script = allowedScripts.find(({ value: name }) => name === answer);
+      console.log(chalk.blue(`\nRunning "${script?.value}"...`));
+      await execShellCommand(`npm run ${script?.value}` + (script?.args ? ` -- ${script.args}` : ''));
+    }
+
+    console.log(chalk.green('\nAll selected scripts completed successfully.'));
+  } catch (error) {
+    if (error instanceof Error) {
+      console.error(chalk.red('Error running scripts:', error.message));
+    }
+    process.exit(1);
+  }
+}
+
+// Function to execute shell commands
```

Comment:
The function adequately describes what it does. No need to add a comment explaining it.

Reply:
Code has been refactored.